Chapter3/src/wci/frontend/java/JavaTokenType.java

```diff
@@ -5,31 +5,34 @@
 
 import wci.frontend.TokenType;
 
-// TODO: Modify reserved words/symbols for Java - this was copied/pasted from PascalTokenType
-
 public enum JavaTokenType implements TokenType {
 	 // Reserved words.
-    AND, ARRAY, BEGIN, CASE, CONST, DIV, DO, DOWNTO, ELSE, END,
-    FILE, FOR, FUNCTION, GOTO, IF, IN, LABEL, MOD,
-    OF, OR, PACKED, PROCEDURE, PROGRAM, RECORD, REPEAT, SET,
-    THEN, TO, TYPE, UNTIL, VAR, WHILE, WITH,
+	ABSTRACT, DOUBLE, INT, LONG, BREAK, ELSE, SWITCH, CASE,
+	ENUM, NATIVE, SUPER, CHAR, EXTENDS, RETURN, THIS, CLASS, FLOAT, SHORT,
+	THROW, CONST, FOR, PACKAGE, VOID, CONTINUE, GOTO, PROTECTED, VOLATILE,
+	DO, IF, STATIC, WHILE,
+
 
     // Special symbols.
-    PLUS("+"), MINUS("-"), STAR("*"), SLASH("/"), DOT("."),
-    COMMA(","), SEMICOLON(";"), COLON(":"), QUOTE("'"),
-    EQUALS("="), EQUALS_CHECK("=="), NOT("!"), NOT_EQUALS("!="), LESS_THAN("<"), LESS_EQUALS("<="),
-    GREATER_EQUALS(">="), GREATER_THAN(">"), LEFT_PAREN("("), RIGHT_PAREN(")"),
-    LEFT_BRACKET("["), RIGHT_BRACKET("]"), LEFT_BRACE("{"), RIGHT_BRACE("}"),
-    UP_ARROW("^"), AND_BIT("&"), PERCENT("%"),
+	BIT_COMPLEMENT("~"), NOT("!"), PERCENT("%"), BIT_XOR("^"), BIT_AND("&"), STAR("*"), BIT_OR("|"),
```

Comment:
I added these as well as I thought this was my responsibility :(
I see you added " // and /* which I don't think belongs in this list as " will be handled by string token and // /* will be treated as whitespace